render-ws-spark-client/src/main/java/org/janelia/render/client/spark/zspacing/ZPositionCorrectionClient.java

```diff
@@ -149,20 +149,25 @@ private void deriveAndSolveCrossCorrelationData(final JavaSparkContext sparkCont
                                    multiProjectParameters,
                                    runName);
 
-        final JavaRDD<StackWithZValues> rddStackWithAllZValues = sparkContext.parallelize(stackWithAllZValuesList);
-
-        final JavaRDD<Integer> rddCompletedSolveCount = rddStackWithAllZValues.map(stackWithZ -> {
-            LogUtilities.setupExecutorLog4j(stackWithZ.toString());
-            final org.janelia.render.client.zspacing.ZPositionCorrectionClient jClient =
-                    clientParameters.buildJavaClient(stackWithZ, comparisonRange, runName, true);
-            final CrossCorrelationData ccData = jClient.loadCrossCorrelationDataSets();
-            jClient.estimateAndSaveZCoordinates(ccData);
-            return 1;
-        });
-
-        final int completedSolveCount = rddCompletedSolveCount.collect().stream().reduce(0, Integer::sum);
+        // only solve (for thickness correction) if requested
+        if (! clientParameters.zSpacing.skipSolve) {
+            final JavaRDD<StackWithZValues> rddStackWithAllZValues = sparkContext.parallelize(stackWithAllZValuesList);
+
+            final JavaRDD<Integer> rddCompletedSolveCount = rddStackWithAllZValues.map(stackWithZ -> {
+                LogUtilities.setupExecutorLog4j(stackWithZ.toString());
+                final org.janelia.render.client.zspacing.ZPositionCorrectionClient jClient =
+                        clientParameters.buildJavaClient(stackWithZ, comparisonRange, runName, true);
+                final CrossCorrelationData ccData = jClient.loadCrossCorrelationDataSets();
+                jClient.estimateAndSaveZCoordinates(ccData);
+                return 1;
```

Comment:
Could this be `forEach()` and `count()` rather than `map()` and `reduce()`?

Reply:
Technically, it can only be foreach - but I agree that is cleaner since the RDD is not needed after the solve.